cl/phase1/network/gossip_manager.go

```diff
@@ -203,6 +213,25 @@ func (g *GossipManager) onRecv(ctx context.Context, data *sentinel.GossipData, l
 			}
 
 			log.Debug("Received blob sidecar via gossip", "index", *data.SubnetId, "size", datasize.ByteSize(len(blobSideCar.Blob)))
+		case gossip.IsTopicBeaconAttestation(data.Name):
+			att := &solid.Attestation{}
+			if err := att.DecodeSSZ(common.CopyBytes(data.Data), int(version)); err != nil {
+				g.sentinel.BanPeer(ctx, data.Peer)
+				l["at"] = "decoding attestation"
+				return err
+			}
+			if err := g.committeeSub.OnReceiveAttestation(data.Name, att); err != nil {
```

Comment:
you probably to move this logic to *forkchoice.ForkChoiceStore, the reason for that is that there is always an up-to date version of the state there. Note that the head state constantly change. this is only for the check of course. what i mean is moving OnReceiveAttestation from committeeSub to the forkchoice object

Reply:
Or you can pass the synced data manager to the `committeeSub` and retrieve the `HeadState()` from there. for ref, look at https://github.com/ledgerwatch/erigon/blob/devel/cl/beacon/synced_data/synced_data.go

Reply:
working on this. but I feel it ForkChoiceStore is getting bigger. maybe we can inverse the dependency between these on_operations and this head state store?  Maybe syncedDataManager is an approach to solve this. 
Anyway I am moving the operation to ForkStore becasue for now it's much cleaner to put on_operation together.

Reply:
oh... sh......t, I found cycle when I moved this function out from committeeSub.
If I move OnReceiveAttestation to forkchoice, then
```
forkChoice import committeeSub, // so can check if needs aggregate
committeeSub import sentinelClient,  // to set expiry when subscribe
sentinelClient import forkChoice 
```
Let's go second approach, import synced data manager : (